views/actions/packages.js

```diff
@@ -79,25 +82,61 @@ export function resetGetPackagesError() {
   };
 }
 
+export function addPackageHistory(history) {
+  return {
+    type: ADD_PACKAGE_HISTORY,
+    history,
+  };
+}
+
 function createGetPackagesRequest() {
   return {
     path: '/packages',
     method: 'GET',
   };
 }
 
+function createGetPackageHistoryRequest(id) {
+  return {
+    path: `/packages/${id}/packageHistory`,
+    method: 'GET',
+  };
+}
+
+function getAndAddPackageHistories(packages) {
+  const historyQueries = packages.map((item) => {
+    const request = createGetPackageHistoryRequest(item.id);
+    return query(request);
+  });
+
+  return Promise.all(historyQueries).then((histories) => {
+    const json = histories.map((data) => {
+      return data.json();
+    });
+    return Promise.all(json);
+  });
+}
+
 export function getSubscribedPackages() {
   const request = createGetPackagesRequest();
+  let packagesResponse;
   return (dispatch, getState) => {
     dispatch(requestPackages());
     query(request).then((response) => {
       response.json().then((json) => {
         if (!response.ok) {
           dispatch(addGetPackagesError(json.message));
         } else {
-          dispatch(receivedPackages(json));
+          packagesResponse = json;
+          return getAndAddPackageHistories(json);
```

Comment:
My thoughts are to use lazy loading. This case is real when a person has many packages. Then we are loading everything and doing a lot of processing especially on package that have already been delivered.